### PR TITLE
Implement EarningsWidget

### PR DESCRIPTION
**Pull request by Autek**
## Description

This PR implements the **EarningsWidget** as part of the Worker HomeScreen in the QuickFix app. The **EarningsWidget** displays key earning statistics for the worker, including:

- Personal balance
- Earnings this month
- Average selling price
- Active orders with their total value

Additionally, this PR includes comprehensive **unit tests** to ensure correct rendering and behavior of the `EarningsWidget`.

---

## Changes Implemented

1. **EarningsWidget Implementation**
   - Displays detailed earning statistics:
     - Personal balance
     - Earnings this month
     - Average selling price
     - Active orders with their respective values
   - Added `testTag`s to key components for easier UI testing.

2. **Unit Tests**
   - Added tests to verify that the `EarningsWidget` renders correctly with sample data.

---

## Screenshot

Capture d’écran 2024-12-17 à 00 17 29

---

### Manual Testing
- Verified that the **EarningsWidget** displays all required data correctly.
- Ensured proper formatting for currency and active order values.
- Confirmed visual alignment and spacing.

### Automated Tests
- Added unit tests for `EarningsWidget`:
   - Verified that all data fields are rendered correctly.
---

## Checklist

- [x] Implement **EarningsWidget** to display earning statistics.
- [x] Add `testTag`s to key components for UI testing.
- [x] Write unit tests for `EarningsWidget`.
- [x] Validate correct data rendering and formatting.
- [x] Achieve at least 80% test coverage.

### CODE REVIEW

**Comment by Autek** [comment]
We decided with the team to only keep the EarningsWidget in the PR so a lot of files are discarded in commit https://github.com/arygm/QuickFix/commit/c743a83bf3e1f7db3525407579584da24c193177

File: app/src/main/java/com/arygm/quickfix/ui/uiMode/appContentUI/workerMode/home/EarningsWidget.kt [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+                .shadow(5.dp, RoundedCornerShape(20.dp))
+                .background(colorScheme.surface, RoundedCornerShape(12.dp))
+                .testTag("EarningsWidget"),
+        verticalArrangement = Arrangement.Top,
+        horizontalAlignment = Alignment.Start) {
+          // Header Row
+          Row(
+              modifier =
+                  Modifier.fillMaxWidth()
+                      .padding(horizontal = 12.dp)
```
Comment (JitterKiller) [inline app/src/main/java/com/arygm/quickfix/ui/uiMode/appContentUI/workerMode/home/EarningsWidget.kt:56]: There's an absolute value here, make sure to use `maxWidth`
```suggestion
                      .padding(horizontal = 12.dp * maxWidth.value)
```

File: app/src/main/java/com/arygm/quickfix/ui/uiMode/appContentUI/workerMode/home/EarningsWidget.kt [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+                      .padding(top = 12.dp)
```
Comment (JitterKiller) [inline app/src/main/java/com/arygm/quickfix/ui/uiMode/appContentUI/workerMode/home/EarningsWidget.kt:57]: Same here, you can use `maxHeight` to remove the use of absolute values.
```suggestion
                      .padding(top = 12.dp * maxHeight.value)
```

File: app/src/main/java/com/arygm/quickfix/ui/uiMode/appContentUI/workerMode/home/EarningsWidget.kt [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+                    fontSize = 19.sp,
+                    fontWeight = FontWeight.Bold,
+                    color = colorScheme.onBackground,
+                    modifier = Modifier.testTag("EarningsTitle"))
+              }
+
+          HorizontalDivider(
+              thickness = 1.dp,
+              color = colorScheme.onSurface.copy(alpha = 0.2f),
+              modifier = Modifier.padding(vertical = 8.dp).testTag("EarningsDivider"))
```
Comment (JitterKiller) [inline app/src/main/java/com/arygm/quickfix/ui/uiMode/appContentUI/workerMode/home/EarningsWidget.kt:73]: Also an absolute value, you can use `maxHeight`

File: app/src/main/java/com/arygm/quickfix/ui/uiMode/appContentUI/workerMode/home/EarningsWidget.kt [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+
+          // Content Rows
+          Row(
+              modifier =
+                  Modifier.fillMaxWidth()
+                      .padding(horizontal = 12.dp)
+                      .padding(bottom = 12.dp)
+                      .testTag("EarningsContent"),
```
Comment (JitterKiller) [inline app/src/main/java/com/arygm/quickfix/ui/uiMode/appContentUI/workerMode/home/EarningsWidget.kt:81]: There are absolute values here

**Review by JitterKiller — CHANGES_REQUESTED** [review]
Great PR !
One thing you could improve is to use `maxWidth` and `maxHeight` from the `BoxWithConstrains` elements to remove the use of absolute values.
Apart from that, the widget looks clean good job on that! 😄

**Comment by Autek** [comment]
Requested changes are done !
DP values are changed for relative value, We should make sure to do the same for the other widgets so that they keep the same comportment !

**Review by JitterKiller — APPROVED** [review]
LGTM!

**Pull request merged**